plugins/modules/terraform.py

```diff
@@ -281,6 +281,14 @@
 import os
 import json
 import tempfile
+
+try:
+    import yaml
+except ImportError as exc:
+    YAML_IMPORT_ERROR = exc
+else:
+    YAML_IMPORT_ERROR = None
```

Comment:
1. You're not handling the case that `YAML_IMPORT_ERROR is not None` (in that case `yaml` cannot be used).
2. You cannot unconditionally add a new requirement, that's a breaking change. So either change the code that diff mode only works when `yaml` is present and document support as `partial` with details (and mention `pyyaml` in the module's requirements), or change the code so it works without PyYAML (no idea if that's possible).

Reply:
In case of missing `pyyaml` I can throw an exception or just return the json diff.

we can discuss this in below conversation. [here](https://github.com/ansible-collections/community.general/pull/7896#discussion_r1468379870)